test/vms.changefeed.test.js

```diff
@@ -0,0 +1,584 @@
+                // Log the state of the nics so that we know why we failed
+                t.deepEqual(nics, {}, 'nics - query: ' + JSON.stringify(query));
+                return callback(new Error('VM does not have valid NICs'));
+            } else {
+                return callback(null, nics[0].state);
+            }
+        });
+    }
+
+    async.doWhilst(
+        function (callback) {
+            getNicStatus(function (err, nicState) {
+                if (err) {
+                    return callback(err);
+                }
+
+                count++;
+                // Assume just one NIC
+                if (nicState === state) {
+                    stop = true;
+                    return callback();
+                } else if (count === maxSeconds) {
+                    stop = true;
+                    return callback(new Error('Timeout waiting on NIC state ' +
+                        'change from ' + nicState + ' to ' + state));
+                }
+
+                setTimeout(callback, 1000);
+            });
+        },
+        function () { return !stop; },
+        waitCallback);
+}
+
+function createOpts(path, params) {
+    return {
+        path: path,
+        headers: {
+            'x-request-id': uuid.create(),
+            'x-context': JSON.stringify({
+                caller: CALLER,
+                params: params || {}
+            })
+        }
+    };
+}
+
+exports.setUp = function (callback) {
+    common.setUp(function (err, _client) {
+        assert.ifError(err);
+        assert.ok(_client, 'restify client');
+        client = _client;
+        callback();
+    });
+};
+
+exports.find_headnode = function (t) {
+    client.cnapi.get('/servers', function (err, req, res, servers) {
+        t.ifError(err);
+        t.equal(res.statusCode, 200);
+        t.ok(servers);
+        t.ok(Array.isArray(servers));
+        for (var i = 0; i < servers.length; i++) {
+            if (servers[i].headnode === true) {
+                SERVER = servers[i];
+                break;
+            }
+        }
+        t.ok(SERVER);
+        t.done();
+    });
+};
+
+
+exports.napi_networks_ok = function (t) {
+    client.napi.get('/networks', function (err, req, res, networks) {
+        t.ifError(err);
+        t.equal(res.statusCode, 200);
+        t.ok(networks);
+        t.ok(Array.isArray(networks));
+        t.ok(networks.length > 1);
+        NETWORKS = networks;
+        t.done();
+    });
+};
+
+exports.create_vm = function (t) {
+    t.expect(8);
+    var md = {
+        foo: 'bar',
+        credentials: JSON.stringify({ 'user_pw': '12345678' })
+    };
+
+    var vm = {
+        owner_uuid: CUSTOMER,
+        image_uuid: IMAGE,
+        server_uuid: SERVER.uuid,
+        networks: [ { uuid: NETWORKS[0].uuid } ],
+        brand: 'joyent-minimal',
+        billing_id: '00000000-0000-0000-0000-000000000000',
+        ram: 64,
+        quota: 10,
+        customer_metadata: md,
+        creator_uuid: CUSTOMER,
```

Comment:
@joshwilsdon This was just to keep consistent with the vms.full.test.js file, which creates its test VMs the same way. I think a second pass on VMAPI tests is in order, after @pfmooney finishes his node-tape PR.